### ISSUE TITLE
Checkbox: "label" should be clickable

### ISSUE THREAD

**Issue by brucelawson** [GoogleChromeLabs/howto-components#112]
Native checkboxes can be toggled by clicking the assocated label (see http://jsbin.com/teroqowumi/edit?html,output). Howto example should behave similarly for a11y and usability.

**Comment by robdodson** [comment]
I agree! Unfortunately the native label doesn't give us that focus behavior because it doesn't support custom elements. I raised this issue with the spec authors. I think we are going to have to write our own howto-label which supports all custom elements. We're on it!

**Comment by robdodson** [comment]
PR for howto-label is here: https://github.com/GoogleChrome/howto-components/pull/114

Still need to add tests and improve the comments but folks can play around with the implementation if they're curious.

**Comment by daKmoR** [comment]
I'm just curious but why is it again totally encapsulated?
I thought a11y wise `paper-input` was a bad idea and it would be better to have something like this
```
<my-input>
  <input id="ni" type="text" ...>
  <label for="ni">First Name</label>
</my-input>
```
why not have the same for checkbox?

I thought I got it figured out and having it in the light dom was the way to go... now I'm confused again :p

**Comment by robdodson** [comment]
Travelling at the moment but will follow up later next week with a longer response. The short answer is that we are following the ARIA Authoring Practices Guide and assuming that developers are going to build custom widgets regardless (perhaps choosing not to wrap native controls) and we want to show them how to do so as responsibly as possible. But I have a much more detailed answer I can add when I get back to a computer 😁

**Comment by daKmoR** [comment]
Thx for answering even while traveling :) have a safe trip :)

Looking forward to a more detailed explanation :)

as u wrote "regardless" I figure u mean ARIA just wants to be prepared for a not easy/optimal usage...? 
So doing it like I wrote before would be as accessable without the "extra" work for my own label implementation right?

For a simpler element I could wrap it again and provide a label as an attribute... to get an "easier" handling for end users.

I hope I'm making sense here :p

I just wanna be sure that adding input + label in the same light or shadow Dom is not a bad idea...

**Comment by daKmoR** [comment]
@robdodson so any update on this?

**Comment by robdodson** [comment]
wrote up an outline of my thoughts last night. it's turning into a blog post :P

**Comment by robdodson** [comment]
Getting ready to publish my post. Also wanted to mention that I've added a warning to the howto-checkbox docs to explain the downsides of using a custom element in this instance. See #121

**Comment by robdodson** [comment]
Finally got this blog post out! https://medium.com/@robdodson/the-future-of-accessibility-for-custom-elements-291cfb3ffabe

**Comment by robdodson** [comment]
This issue should be fixed when #124 lands
